src/map/script.c

```diff
@@ -12865,42 +12923,47 @@ static BUILDIN(getwaitingroomstate)
 {
 	const struct npc_data *nd;
 	const struct chat_data *cd;
-	int type;
+	int type = script_getnum(st, 2);
 	int i;
 
-	type = script_getnum(st,2);
-	if( script_hasdata(st,3) )
+	if (script_hasdata(st, 3)) {
 		nd = npc->name2id(script_getstr(st, 3));
-	else
+		if (nd == NULL) {
+			ShowWarning("buildin_getwaitingroomstate: NPC '%s' not found.\n", script_getstr(st, 3));
+			return false;
+		}
+	} else
 		nd = map->id2nd(st->oid);
 
-	if (nd == NULL || (cd=map->id2cd(nd->chat_id)) == NULL) {
+	if (nd == NULL || (cd = map->id2cd(nd->chat_id)) == NULL) {
 		script_pushint(st, -1);
 		return true;
 	}
 
-	switch(type) {
-		case 0:
-			for (i = 0; i < cd->users; i++) {
-				struct map_session_data *sd = cd->usersd[i];
-				nullpo_retr(false, sd);
-				mapreg->setreg(reference_uid(script->add_variable("$@chatmembers"), i), sd->bl.id);
-			}
-			script_pushint(st, cd->users);
-			break;
-		case 1:  script_pushint(st, cd->limit); break;
-		case 2:  script_pushint(st, cd->trigger&0x7f); break;
-		case 3:  script_pushint(st, ((cd->trigger&0x80)!=0)); break;
-		case 4:  script_pushstrcopy(st, cd->title); break;
-		case 5:  script_pushstrcopy(st, cd->pass); break;
-		case 16: script_pushstrcopy(st, cd->npc_event);break;
-		case 32: script_pushint(st, (cd->users >= cd->limit)); break;
-		case 33: script_pushint(st, (cd->users >= cd->trigger)); break;
-
-		case 34: script_pushint(st, cd->min_level); break;
-		case 35: script_pushint(st, cd->max_level); break;
-		case 36: script_pushint(st, cd->zeny); break;
-		default: script_pushint(st, -1); break;
+	switch (type) {
+	case 0:
+		for (i = 0; i < cd->users; i++) {
+			struct map_session_data *sd = cd->usersd[i];
+			nullpo_retr(false, sd);
+			mapreg->setreg(reference_uid(script->add_variable("$@chatmembers"), i), sd->bl.id);
+		}
+		script_pushint(st, cd->users);
+		break;
+	case 1:  script_pushint(st, cd->limit); break;
+	case 2:  script_pushint(st, cd->trigger & 0x7f); break;
+	case 3:  script_pushint(st, ((cd->trigger & 0x80) != 0)); break;
+	case 4:  script_pushstrcopy(st, cd->title); break;
+	case 5:  script_pushstrcopy(st, cd->pass); break;
+	case 16: script_pushstrcopy(st, cd->npc_event); break;
+	case 32: script_pushint(st, (cd->users >= cd->limit)); break;
+	case 33: script_pushint(st, (cd->users >= cd->trigger)); break;
+	case 34: script_pushint(st, cd->min_level); break;
+	case 35: script_pushint(st, cd->max_level); break;
+	case 36: script_pushint(st, cd->zeny); break;
+	default: 
+		script_pushint(st, -1);
+		ShowWarning("buildin_getwaitingroomstate: invalid type '%s'.\n", type);
```

Comment:
```c
prontera,155,185,5	script	kjdsfhskdf	1_F_MARIA,{
	dispbottom getwaitingroomstate(6)+ "";
	end;
OnInit:
	waitingroom "asdf", 0;
}
```
map-server crash on this line
```
[Warning]:
Map-Server has shutdown successfully.
Restarting in 15 seconds, press Ctrl+C to cancel.
```

```suggestion
		ShowWarning("buildin_getwaitingroomstate: invalid type '%d'.\n", type);
```

Reply:
done